internal/test/factory/p2p.go

```diff
@@ -0,0 +1,25 @@
+package factory
```

Comment:
Is this a new package? If so can it please get a package doc?

Reply:
This is used for internal testing. Do you think it still needs a package doc?

Reply:
IMO it should, even if only to mention that fact, e.g.,

```go
// Package factory provides custom node identifier generation code
// for internal testing of the state sync mechanism.
```

or something like that would be fine. Rationale: Even though it's an internal package, it's still publicly visible, and it helps for debugging if even test/support packages have some context in godoc.org.

I would support documenting the exported methods in test/support packages on that basis, too. That's also a good place to call out "this is a testing thing, not subject to compatibility promises, don't use this or your pants will fall down", etc.